explorer/ast/pattern.h

```diff
@@ -17,7 +17,8 @@
 #include "explorer/ast/value_category.h"
 #include "explorer/common/source_location.h"
 #include "llvm/ADT/ArrayRef.h"
-#include "llvm/ADT/STLFunctionalExtras.h"
+//the include bellow doesn't seem to be used and doesn't exists on clang13
+//#include "llvm/ADT/STLFunctionalExtras.h"
```

Comment:
Please revert these changes.

Two things:

1) The installed LLVM compiler is separate from what we use for libraries. [Libraries are hermetic](https://github.com/carbon-language/carbon-lang/blob/trunk/WORKSPACE#L98-L128).

2) In case you're using clang-13 for compilation, please do note we suggest [clang-14 as minimum](https://github.com/carbon-language/carbon-lang/blob/trunk/docs/project/contribution_tools.md#old-llvm-versions).

Reply:
I only noticed that `#include "llvm/ADT/STLFunctionalExtras.h"` isn't used because I tested it with clang-13, fell free to remove it entirelly.